dataconnect/dataconnect/movie-connector/queries.gql

```diff
@@ -0,0 +1,363 @@
+query ListMoviesByTag($tag: String!) @auth(level: PUBLIC) {
+  movies(where: { tags: { includes: $tag } }) {
+    id
+    title
+    imageUrl
+    genre
+    rating
+  }
+}
+
+# List movies by release year range
+query MoviesByReleaseYear($min: Int, $max: Int) @auth(level: PUBLIC) {
+  movies(
+    where: { releaseYear: { le: $max, ge: $min } }
+    orderBy: { releaseYear: ASC }
+  ) {
+    id
+    rating
+    title
+    imageUrl
+  }
+}
+
+# List movies by rating and genre with OR filters
+query SearchMovieOr(
+  $minRating: Float
+  $maxRating: Float
+  $genre: String
+  $tag: String
+  $input: String
+) @auth(level: PUBLIC) {
+  movies(
+    where: {
+      _or: [
+        { rating: { ge: $minRating } }
+        { rating: { le: $maxRating } }
+        { genre: { eq: $genre } }
+        { tags: { includes: $tag } }
+        { title: { contains: $input } }
+      ]
+    }
+  ) {
+    id
+    rating
+    title
+    imageUrl
+  }
+}
+
+# List movies by rating and genre with AND filters
+query SearchMovieAnd(
+  $minRating: Float
+  $maxRating: Float
+  $genre: String
+  $tag: String
+  $input: String
+) @auth(level: PUBLIC) {
+  movies(
+    where: {
+      _and: [
+        { rating: { ge: $minRating } }
+        { rating: { le: $maxRating } }
+        { genre: { eq: $genre } }
+        { tags: { includes: $tag } }
+        { title: { contains: $input } }
+      ]
+    }
+  ) {
+    id
+    rating
+    title
+    imageUrl
+  }
+}
+
+# Get favorite movies by user ID
+query GetUserFavoriteMovies @auth(level: USER) {
+  user(id_expr: "auth.uid") {
+    favorite_movies_on_user {
+      movie {
+        id
+        title
+        genre
+        imageUrl
+        releaseYear
+        rating
+        description
+      }
+    }
+  }
+}
+
+# Get favorite actors by user ID
+query GetFavoriteActors @auth(level: USER) {
+  user(key: {id_expr: "auth.uid"}) {
+    favorite_actors_on_user {
+      actor {
+        id
+        name
+        imageUrl
+      }
+    }
+  }
+}
+
+# Get user by ID
+query GetCurrentUser @auth(level: USER) {
+  user(key: {id_expr: "auth.uid"}) {
+    id
+    username
+    reviews: reviews_on_user {
+      id
+      rating
+      reviewDate
+      reviewText
+      movie {
+        id
+        title
+      }
+    }
+    favoriteMovies: favorite_movies_on_user {
+      movie {
+        id
+        title
+        genre
+        imageUrl
+        releaseYear
+        rating
+        description
+        tags
+        metadata: movieMetadatas_on_movie {
+          director
+        }
+      }
+    }
+    favoriteActors: favorite_actors_on_user {
+      actor {
+        id
+        name
+        imageUrl
+      }
+    }
+  }
+}
+
+# Check if a movie is favorited by user
+query GetIfFavoritedMovie($movieId: UUID!) @auth(level: USER) {
+  favorite_movie(key: { userId_expr: "auth.uid", movieId: $movieId }) {
+    movieId
+  }
+}
+
+# Check if an actor is favorited by user
+query GetIfFavoritedActor($actorId: UUID!) @auth(level: USER) {
```

Comment:
Would you mind commenting which ones are unused and used as an example? You could do an overall blocked comment like:
```
# the queries below are unused by the application, but are useful examples for more complex cases
query1...
query2...
# end of example queries
```